src/transformation/mesh_intersection/mesh_intersection.rs

```diff
@@ -256,380 +355,358 @@ fn extract_connected_components(
+                    .map(|tri| {
+                        SimplePolygon(vec![
+                            IndexTuple(tri[0] as usize, None, None),
+                            IndexTuple(tri[1] as usize, None, None),
+                            IndexTuple(tri[2] as usize, None, None),
+                        ])
+                    })
+                    .collect(),
+                name: "".to_string(),
+                index: 0,
+                material: None,
+            }],
+            name: "".to_string(),
+        }],
+        ..Default::default()
+    }
+    .write_to_buf(&mut file)
+    .unwrap();
 }
 
-impl Triangulation {
-    fn new(triangle: Triangle) -> Self {
-        let mut delaunay = ConstrainedDelaunayTriangulation::<spade::Point2<Real>>::new();
-        let normal = triangle.normal().unwrap();
-        let basis = normal.orthonormal_basis();
-
-        let ab = triangle.b - triangle.a;
-        let ac = triangle.c - triangle.a;
-
-        let mut ref_proj = [
-            Point2::origin(),
-            Point2::new(ab.dot(&basis[0]), ab.dot(&basis[1])),
-            Point2::new(ac.dot(&basis[0]), ac.dot(&basis[1])),
-        ];
-
-        let normalization_inv =
-            na::Matrix2::from_columns(&[ref_proj[1].coords, ref_proj[2].coords]);
-        let normalization = normalization_inv
-            .try_inverse()
-            .unwrap_or(na::Matrix2::identity());
-
-        for ref_proj in &mut ref_proj {
-            *ref_proj = normalization * *ref_proj;
-        }
+// I heavily recommend that this is left here in case one needs to debug the above code.
+fn _points_to_obj(mesh: &[Point3<f64>], path: &PathBuf) {
+    use std::io::Write;
+    let mut file = std::fs::File::create(path).unwrap();
 
-        let vtx_handles = [
-            delaunay
-                .insert(spade::Point2::new(ref_proj[0].x, ref_proj[0].y))
-                .unwrap(),
-            delaunay
-                .insert(spade::Point2::new(ref_proj[1].x, ref_proj[1].y))
-                .unwrap(),
-            delaunay
-                .insert(spade::Point2::new(ref_proj[2].x, ref_proj[2].y))
-                .unwrap(),
-        ];
+    for p in mesh {
+        writeln!(file, "v {} {} {}", p.x, p.y, p.z).unwrap();
+    }
+}
 
-        Self {
-            delaunay,
-            basis,
-            vtx_handles,
-            ref_pt: triangle.a,
-            ref_proj,
-            normalization,
+// I heavily recommend that this is left here in case one needs to debug the above code.
+fn _points_and_edges_to_obj(mesh: &[Point3<f64>], edges: &[[usize; 2]], path: &PathBuf) {
+    use std::io::Write;
+    let mut file = std::fs::File::create(path).unwrap();
+
+    for p in mesh {
+        writeln!(file, "v {} {} {}", p.x, p.y, p.z).unwrap();
+    }
+
+    for e in edges {
+        writeln!(file, "l {} {}", e[0] + 1, e[1] + 1).unwrap();
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Debug, Default)]
+struct TreePoint {
+    point: Vector3<f64>,
```

Comment:
A field named `point` surely has to be a `Point3`, not a vector.

Reply:
A point is a vector! But I will change it T.T

Reply:
Sorry, but a point is just not the same as a vector. The fact that you called the variable `point` instead of `vector` shows that, through the name of the argument, you wanted to give semantic to that vector. A `Point3` gives you that semantic at the type level.

Reply:
See my comment above. I called it a point because the examples on the library I used for the rtree named it point and I just copy pasted that example and edited accordingly.

On the latest branch, all variables have been renamed according to the specification, i.e. they are all `Point3`.